cmd/routesrv/main.go

```diff
@@ -0,0 +1,262 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	ot "github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/cmd/routesrv/options"
+	"github.com/zalando/skipper/config"
+	"github.com/zalando/skipper/dataclients/kubernetes"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/routing"
+	"github.com/zalando/skipper/tracing"
+)
+
+// eskipBytes keeps eskip-formatted routes as a byte slice and
+// provides synchronized r/w access to them. Additionally it can
+// serve as an HTTP handler exposing its content.
+type eskipBytes struct {
+	data        []byte
+	mu          sync.RWMutex
+	initialized bool
+
+	tracer ot.Tracer
+}
+
+func (e *eskipBytes) bytes() ([]byte, bool) {
+	e.mu.RLock()
+	defer e.mu.RUnlock()
+
+	return e.data, e.initialized
+}
+
+func (e *eskipBytes) formatAndSet(routes []*eskip.Route) (int, bool) {
+	buf := &bytes.Buffer{}
+	eskip.Fprint(buf, eskip.PrettyPrintInfo{}, routes...)
+
+	e.mu.Lock()
+	oldData := e.data
+	e.data = buf.Bytes()
+	e.initialized = true
+	e.mu.Unlock()
+
+	return len(oldData), oldData == nil
```

Comment:
WDYT if we return (newData, oldData []byte) so that users can take length or detect the initialization by checking `oldData == nil` themselves. This also eliminates `initialized` field.

Reply:
Makes sense.